<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Stephanie Nguyen">
<meta name="dcterms.date" content="2023-06-06">

<title>Final homework Write Up</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="final_homework_draft_files/libs/clipboard/clipboard.min.js"></script>
<script src="final_homework_draft_files/libs/quarto-html/quarto.js"></script>
<script src="final_homework_draft_files/libs/quarto-html/popper.min.js"></script>
<script src="final_homework_draft_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="final_homework_draft_files/libs/quarto-html/anchor.min.js"></script>
<link href="final_homework_draft_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="final_homework_draft_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="final_homework_draft_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="final_homework_draft_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="final_homework_draft_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
  <li><a href="#set-up" id="toc-set-up" class="nav-link" data-scroll-target="#set-up">Set Up</a></li>
  </ul></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">Methods</a>
  <ul class="collapse">
  <li><a href="#full-model" id="toc-full-model" class="nav-link" data-scroll-target="#full-model">Full Model</a></li>
  <li><a href="#full-log-model" id="toc-full-log-model" class="nav-link" data-scroll-target="#full-log-model">Full Log Model</a></li>
  <li><a href="#additional-models" id="toc-additional-models" class="nav-link" data-scroll-target="#additional-models">Additional Models</a></li>
  <li><a href="#check-aic-of-models" id="toc-check-aic-of-models" class="nav-link" data-scroll-target="#check-aic-of-models">Check AIC of models</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a>
  <ul class="collapse">
  <li><a href="#data-dictionary" id="toc-data-dictionary" class="nav-link" data-scroll-target="#data-dictionary">Data Dictionary</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Final homework Write Up</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Stephanie Nguyen </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 6, 2023</p>
    </div>
  </div>
    
  </div>
  

</header>

<p>Link to Github repo: https://github.com/stephanie-t-nguyen/ENVS-193DS_homework-05</p>
<p>Link to rendered HTML: https://stephanie-t-nguyen.github.io/ENVS-193DS_homework-05/code/final_homework_draft.html</p>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>The <em>Sarracena</em> genus are a group of North American carnivorous plants that have radiated into a number of similar but ultimately different species <span class="citation" data-cites="stephens2015">(<a href="#ref-stephens2015" role="doc-biblioref">Stephens et al. 2015</a>)</span>. They are characterized by their pitcher shape structures which lures insects and thus are known as the pitcher plants. They typically grow in nitrogen-low and nutrient-poor soils, which includes bogs, wetlands, and fens <span class="citation" data-cites="srivastava2011">(<a href="#ref-srivastava2011" role="doc-biblioref">Srivastava et al. 2011</a>)</span>.</p>
<p><em>Sarracena</em> are studied for a number of reasons. <em>Sarracena</em> play a very important ecological role in their ecosystems and food webs and therefore could be an indicator of an environment’s health <span class="citation" data-cites="hoekman2010">(<a href="#ref-hoekman2010" role="doc-biblioref">Hoekman 2010</a>)</span>. Our study aims to investigate how morphological, physiological, and/or taxonomic <em>Sarracenia</em> characteristics predict total plant biomass. Being able to predict plant biomass from <em>Sarracenia</em> characteristics allows for a faster and non-destructive method to assess the development of the plant, and in extension, ecosystem health.</p>
<p>Using the data obtained from “Effects of Prey Availability on Sarracena Physiology”, our study investigates how <em>Sarracenia</em> characteristics predict biomass <span class="citation" data-cites="ellison2021">(<a href="#ref-ellison2021" role="doc-biblioref">Ellison and Farnsworth 2021</a>)</span>. We hypothesize that <em>Sarracenia</em> species, specific leaf area, chlorophyll levels, photosynthetic rate, and/or number of pitchers and phyllodes produced by the plant to be predictors of total plant biomass using multiple linear regression.</p>
<section id="set-up" class="level3">
<h3 class="anchored" data-anchor-id="set-up">Set Up</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">message =</span> <span class="cn">FALSE</span>, <span class="at">warning =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggeffects)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(performance)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(naniar)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flextable)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AICcmodavg)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GGally)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>plant <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"knb-lter-hfr.109.18"</span>, <span class="st">"hf109-01-sarracenia.csv"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">clean_names</span>() <span class="sc">|&gt;</span> <span class="co">#Clean column names</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(totmass, species, feedlevel, sla, chlorophyll, amass, num_lvs, num_phylls) <span class="co">#Select relevant columns</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="methods" class="level1">
<h1>Methods</h1>
<p>a. Ellison et al.&nbsp;made 6 feeding levels, accounting for the sizes of different species, and assigned two of each <em>Sarracenia</em> species to each level <span class="citation" data-cites="ellison2021">(<a href="#ref-ellison2021" role="doc-biblioref">Ellison and Farnsworth 2021</a>)</span>. Total plant biomass, feeding level of wasps, specific leaf area, chlorophyll, total number of phyllodes, and total number of pitchers were measured after the feeding period, while photosynthetic rate was measured prior to treatment <span class="citation" data-cites="ellison2021">(<a href="#ref-ellison2021" role="doc-biblioref">Ellison and Farnsworth 2021</a>)</span>.</p>
<p>b. The data from “Effects of Prey Availability on Sarracenia Physiology” recorded these physiological, morphological, and taxonomic characteristics of the <em>Sarracena</em> into a dataframe called plants. Refer to data dictionary at the bottom of the page for explanations of column names</p>
<p>c. Then, the dataframe was checked for missing data (Figure 1). Some data points were missing, so rows with missing data were dropped, resulting in a new dataframe called plant_subset. This is the dataset we used going forward with the analysis.</p>
<p>d. In figure 2, Pearson’s correlation coefficients were calculated between every numerical predictor variable. There were a mix of positive and negative correlations. However, none of the correlation values had an absolute value above 0.5, meaning they were all weakly correlated.</p>
<p>e. To further investigate the relationships between variables, a pair plot was created in figure 3. Figure 3 denotes that there are significant, although weak, correlations between some predictor variables, which is denoted with asterisks.</p>
<p>f.</p>
<p><br>
<br>
f.&nbsp;Creating the null and full model with accompanying discussion (1-3 sentences)<br>
g. Visual and statistical assumption checks for the full model with accompanying<br>
discussion (1-3 sentences)<br>
h. Explanation of any transformation and justification (1-3 sentences)<br>
i. Model construction with visual and statistical assumption checks for three<br>
additional models with justification (i.e.&nbsp;why did you choose those predictor<br>
variables?) (justification: 1-3 sentences per model)<br>
j. Variance inflation factor check for the full model with accompanying discussion<br>
(discussion: 1-3 sentences)<br>
k. Model comparison using the null model, full model, and additional models from part<br>
7g</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gg_miss_var</span>(plant) <span class="co">#Visualize missing data</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="final_homework_draft_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 1.</strong> Visualization of Missing Data in “plant” dataframe</p>
<p>The y-axis are the variables (columns) in the dataframe and the x-axis marks how many rows have missing data per variable.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Drop missing data</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>plant_subset <span class="ot">&lt;-</span> plant <span class="sc">|&gt;</span> </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(sla, chlorophyll, amass, num_lvs, num_phylls)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate Pearson's r for numerical values only </span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>plant_cor <span class="ot">&lt;-</span> plant_subset <span class="sc">|&gt;</span> </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(feedlevel<span class="sc">:</span>num_phylls) <span class="sc">|&gt;</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cor</span>(<span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Create correlation plot</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(plant_cor, </span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">method =</span> <span class="st">"ellipse"</span>, <span class="co">#Change shape of item in cells</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">addCoef.col =</span> <span class="st">"black"</span>) <span class="co">#Add coefficient in black text</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="final_homework_draft_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 2.</strong> Pearson’s correlation plot of predictor variables</p>
<p>The positively correlated predictor variables are colored in blue and Pearson’s correlation values of greater than 0, while the negatively correlated predictor variables are colored in red and have a coefficient of lesser than 0.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>plant_subset <span class="sc">|&gt;</span> </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(species<span class="sc">:</span>num_phylls) <span class="sc">|&gt;</span> </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggpairs</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="final_homework_draft_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 3.</strong> Pair plot of predictor variables</p>
<p>Above the diagonal are Pearson correlation coefficients between the predictor variables, with asterisks indicating a significant correlation. Along the diagonal are density plots for each variable, and below the diagonal are the scatter plots for each variable.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Create a null and a full model - fit multiple linear models in order to see which one is best</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>null <span class="ot">&lt;-</span> <span class="fu">lm</span>(totmass <span class="sc">~</span> <span class="dv">1</span>, <span class="at">data =</span> plant_subset)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Assume no relationship between total mass and predictor variables </span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>full <span class="ot">&lt;-</span> <span class="fu">lm</span>(totmass <span class="sc">~</span> species <span class="sc">+</span> feedlevel <span class="sc">+</span> sla <span class="sc">+</span> chlorophyll <span class="sc">+</span> amass <span class="sc">+</span> num_lvs <span class="sc">+</span> num_phylls, <span class="at">data =</span> plant_subset)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Assume predictor variables can predict biomass</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="full-model" class="level3">
<h3 class="anchored" data-anchor-id="full-model">Full Model</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(full)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="final_homework_draft_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 4.</strong> Diagnostic plots for full model</p>
<p>The Residual vs Fitted plot checks for constant variance (homoscedasticity) among the residuals. The Scale-Location plot also checks for homoscedasticity, but using the square root of the residuals; both plots show a change in pattern from more clustered points on the left to more scattered points on the right, suggesting heteroscedasticty.</p>
<p>The Residuals vs Fitted checks for constant variance of residuals, but the clustering of points on the left side of the plot indicators non-homoscedastic residuals, while the Normal Q-Q checks for normality but the tail ends suggest lack of normality. The Scale-Location shows the square root of the standardized residuals, which also suggests heteroscedastic residuals because there is a pattern of clustering to the spreading of points from left to right. Lastly, the Residuals vs Leverage plot shows outliers in the data, and there do not appear to be any.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">check_normality</span>(full) <span class="co">#The data is likely not normal. </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Warning: Non-normality of residuals detected (p &lt; .001).</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">check_heteroscedasticity</span>(full) <span class="co">#The data is likely heteroscedastic. </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Warning: Heteroscedasticity (non-constant error variance) detected (p &lt; .001).</code></pre>
</div>
</div>
</section>
<section id="full-log-model" class="level3">
<h3 class="anchored" data-anchor-id="full-log-model">Full Log Model</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Log transformations of null and full model</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>null_log <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="fu">log</span>(totmass) <span class="sc">~</span> <span class="dv">1</span>, <span class="at">data =</span> plant_subset)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>full_log <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="fu">log</span>(totmass) <span class="sc">~</span> species <span class="sc">+</span> feedlevel <span class="sc">+</span> sla <span class="sc">+</span> chlorophyll <span class="sc">+</span> amass <span class="sc">+</span> num_lvs <span class="sc">+</span> num_phylls, <span class="at">data =</span> plant_subset)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(full_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="final_homework_draft_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 5.</strong> Diagnostic plots for full log model</p>
<p>The Residual vs Fitted and Scale-Location plot checks for homoscedasticity; both plots don’t show can non-random patterning, indicating homoscedasticity. The Normal Q-Q plot checks for data normality; the data fits the linear line indicating normality. Lastly, the Residuals vs Leverage checks for outliers outside Cook’s distance, but there are no outliers.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">check_normality</span>(full_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>OK: residuals appear as normally distributed (p = 0.107).</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">check_heteroscedasticity</span>(full_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>OK: Error variance appears to be homoscedastic (p = 0.071).</code></pre>
</div>
</div>
<p>Evaluate multicollinearity:</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>car<span class="sc">::</span><span class="fu">vif</span>(full_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                 GVIF Df GVIF^(1/(2*Df))
species     42.351675  9        1.231351
feedlevel    1.621993  1        1.273575
sla          1.999989  1        1.414210
chlorophyll  1.949828  1        1.396362
amass        2.872084  1        1.694722
num_lvs      2.813855  1        1.677455
num_phylls   2.995510  1        1.730754</code></pre>
</div>
</div>
<p>We evaluated multicollinearity by calculating generalized variance inflation factor and determined that…</p>
<p>try some more models:</p>
<p>addressing the question: what set of predictor variables best explains the response?</p>
</section>
<section id="additional-models" class="level3">
<h3 class="anchored" data-anchor-id="additional-models">Additional Models</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Create other models </span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>model2_log <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="fu">log</span>(totmass) <span class="sc">~</span> species, <span class="at">data =</span> plant_subset) <span class="co">#Log total mass, predictor variable is species </span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>model3_log <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="fu">log</span>(totmass) <span class="sc">~</span> species <span class="sc">+</span> feedlevel, <span class="at">data =</span> plant_subset) <span class="co">#Log total mass, predictor variable is species and feedlevel</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>model4_log <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="fu">log</span>(totmass) <span class="sc">~</span> species <span class="sc">+</span> feedlevel <span class="sc">+</span> num_lvs, <span class="at">data =</span> plant_subset) <span class="co">#Log total mass, predictor variable is species, feedlevel, and number of pitchers and phyllodes produced by plant </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We evaluated multicollinearity by calculating generalized variance inflation factor and determined that…</p>
<p>try some more models:</p>
<p>addressing the question: what set of predictor variables best explains the response?</p>
<p>check assumptions for model 2:</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(model2_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="final_homework_draft_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">check_normality</span>(model2_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>OK: residuals appear as normally distributed (p = 0.374).</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">check_heteroscedasticity</span>(model2_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>OK: Error variance appears to be homoscedastic (p = 0.100).</code></pre>
</div>
</div>
</section>
<section id="check-aic-of-models" class="level3">
<h3 class="anchored" data-anchor-id="check-aic-of-models">Check AIC of models</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">AICc</span>(full_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 133.9424</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">AICc</span>(model2_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 157.5751</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">AICc</span>(null_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 306.0028</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>MuMIn<span class="sc">::</span><span class="fu">AICc</span>(full_log, model2_log, null_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>           df     AICc
full_log   17 133.9424
model2_log 11 157.5751
null_log    2 306.0028</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>MuMIn<span class="sc">::</span><span class="fu">model.sel</span>(full_log, model2_log, null_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Model selection table 
             (Int)      ams      chl     fdl num_lvs  num_phy       sla spc df
full_log   -1.3390 0.002338 0.004368 -0.4743 0.09176 -0.03959 -0.002493   + 17
model2_log  0.8836                                                        + 11
null_log    1.3500                                                           2
             logLik  AICc  delta weight
full_log    -46.371 133.9   0.00      1
model2_log  -66.337 157.6  23.63      0
null_log   -150.941 306.0 172.06      0
Models ranked by AICc(x) </code></pre>
</div>
</div>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<p>A description of the best model and how you chose it, with a summary of model<br>
results (parenthetical and/or table, 1-3 sentences)<br>
b. Interpretation of your chosen model (1-3 sentences)<br>
c.&nbsp;Visualization of model predictions for biomass as a function of the predictor<br>
variable of your choice with an accompanying caption (caption: 1-3 sentences)<br>
d.&nbsp;Discussion of results: what does this model mean biologically? (1-3 sentences)</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(full_log)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = log(totmass) ~ species + feedlevel + sla + chlorophyll + 
    amass + num_lvs + num_phylls, data = plant_subset)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.88872 -0.20811  0.02825  0.24218  0.78287 

Coefficients:
                    Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)        -1.339043   0.597727  -2.240 0.027624 *  
speciesalata        1.113163   0.184021   6.049 3.56e-08 ***
speciesflava        1.404562   0.262955   5.341 7.29e-07 ***
speciesjonesii      0.319652   0.196426   1.627 0.107281    
speciesleucophylla  1.709035   0.227608   7.509 4.88e-11 ***
speciesminor        0.389310   0.187903   2.072 0.041239 *  
speciespsittacina  -1.645198   0.207035  -7.946 6.36e-12 ***
speciespurpurea    -0.364348   0.254380  -1.432 0.155643    
speciesrosea       -0.947383   0.260495  -3.637 0.000467 ***
speciesrubra        0.875342   0.196361   4.458 2.46e-05 ***
feedlevel          -0.474255   0.234493  -2.022 0.046199 *  
sla                -0.002493   0.001160  -2.149 0.034430 *  
chlorophyll         0.004368   0.001189   3.672 0.000414 ***
amass               0.002338   0.002988   0.782 0.436166    
num_lvs             0.091764   0.022413   4.094 9.46e-05 ***
num_phylls         -0.039585   0.051714  -0.765 0.446068    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.413 on 87 degrees of freedom
Multiple R-squared:  0.8687,    Adjusted R-squared:  0.8461 
F-statistic: 38.38 on 15 and 87 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
<section id="data-dictionary" class="level3">
<h3 class="anchored" data-anchor-id="data-dictionary">Data Dictionary</h3>
<p>Below is a data dictionary for variables in the plants dataframe.</p>
<ul>
<li><p>totmass = total plant biomass</p></li>
<li><p>species = what species of pitcher plant within the genus Sarracenia</p></li>
<li><p>feedlevel = grams of wasp fed per week</p></li>
<li><p>sla = specific leaf area in cm2/g</p></li>
<li><p>chlorophyll = chlorophyll context in mg/g</p></li>
<li><p>amass = photosynthetic rate in nmol CO2 x g-1 x s-1</p></li>
<li><p>num_lvs = number of phyllodes AND pitchers</p></li>
<li><p>num_phylls = number of phyllodes</p></li>
</ul>

</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-ellison2021" class="csl-entry" role="doc-biblioentry">
Ellison, Aaron, and Elizabeth Farnsworth. 2021. <span>“Effects of Prey Availability on Sarracenia Physiology at Harvard Forest 2005.”</span> Environmental Data Initiative. <a href="https://doi.org/10.6073/PASTA/26B22D09279E62FD729FFC35F9EF0174">https://doi.org/10.6073/PASTA/26B22D09279E62FD729FFC35F9EF0174</a>.
</div>
<div id="ref-hoekman2010" class="csl-entry" role="doc-biblioentry">
Hoekman, David. 2010. <span>“Relative Importance of Top-down and Bottom-up Forces in Food Webs of Sarracenia Pitcher Communities at a Northern and a Southern Site.”</span> <em>Oecologia</em> 165 (4): 1073–82. <a href="https://doi.org/10.1007/s00442-010-1802-2">https://doi.org/10.1007/s00442-010-1802-2</a>.
</div>
<div id="ref-srivastava2011" class="csl-entry" role="doc-biblioentry">
Srivastava, A., W. L. Rogers, C. M. Breton, L. Cai, and R. L. Malmberg. 2011. <span>“Transcriptome Analysis of Sarracenia, an Insectivorous Plant.”</span> <em>DNA Research</em> 18 (4): 253–61. <a href="https://doi.org/10.1093/dnares/dsr014">https://doi.org/10.1093/dnares/dsr014</a>.
</div>
<div id="ref-stephens2015" class="csl-entry" role="doc-biblioentry">
Stephens, Jessica D., Willie L. Rogers, Karolina Heyduk, Jennifer M. Cruse-Sanders, Ron O. Determann, Travis C. Glenn, and Russell L. Malmberg. 2015. <span>“Resolving Phylogenetic Relationships of the Recently Radiated Carnivorous Plant Genus Sarracenia Using Target Enrichment.”</span> <em>Molecular Phylogenetics and Evolution</em> 85 (April): 76–87. <a href="https://doi.org/10.1016/j.ympev.2015.01.015">https://doi.org/10.1016/j.ympev.2015.01.015</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>